src/types/routeMethods.spec-d.ts

```diff
@@ -83,7 +82,7 @@ test('matches individual optional params', () => {
 
   expectTypeOf(router.routes.foo).parameter(0).toEqualTypeOf<{
     foo?: string,
-  }>()
+  } | undefined>()
```

Comment:
Fixing the RouteMethod type caught that these tests were passing but incorrect. All params are optional so the params argument itself is optional as well.